tests/form_params_test.go

```diff
@@ -0,0 +1,385 @@
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+		},
+		{
+			name: "Integer Overflow Content-Length",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" +
+				"Content-Length: 10000000000000000000000000\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), test.req, func(req *http.Request) {
+				t.Error("MakeRequest(): expected handler not to be called.")
+			})
+			if err != nil {
+				t.Fatalf("MakeRequest(): got err %v, want nil", err)
+			}
+			if !bytes.HasPrefix(resp, []byte(statusBadReq)) {
+				t.Errorf("response status: got %s, want %s", string(resp), statusBadReq)
+			}
+		})
+	}
+}
+
+// Tests behaviour when multiple Content-Length values are passed. Go's
+// behaviour is the following:
+// a) if we pass two Content-Length headers containing the same value, the // request
+// is deemed valid
+// b) if we pass two Content-Length headers containing different values, the
+// server will respond with a 400 Bad Request
+// c) if we pass one Content-Length header containing a list of equal values,
+// the server will respond with a 400 Bad Request
+// For a) and c), RFC 7320, Section 3.3.2 says that the server can either accept
+// the request and use the duplicated value or reject it. However, a more
+// consistent behaviour would be to handle a) and c) similarly.
+func TestFormParametersDuplicateContentLength(t *testing.T) {
+	type want struct {
+		contentLen    string
+		queryParamVal string
+	}
+	var tests = []struct {
+		name string
+		req  []byte
+		want want
+	}{
+		{
+			name: "Two Content-Length Headers",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" +
+				"Content-Length: 13\r\n" +
+				"Content-Length: 13\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+			want: want{
+				contentLen:    "13",
+				queryParamVal: "potato",
+			},
+		},
+		{
+			name: "List of same value in Content-Length Header",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" +
+				"Content-Length: 13, 13\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+			want: want{
+				contentLen:    "13",
+				queryParamVal: "potato",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		// Skipping to ensure GitHub checks are not failing. This test is made
+		// to fail intentionally to pinpoint behaviour we regard as inconsistent
+		t.Skip()
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), test.req, func(req *http.Request) {
+				if contentLen := req.Header["Content-Length"][0]; contentLen != test.want.contentLen {
```

Comment:
Use `cmp` for the whole `req.Header["Content-Length"]` slice. Here and in other places.

Reply:
Resolved in [a523171](url)